### PR TITLE
[SERV-600] Enforce camelCase in Database JSON View

### PR DESCRIPTION
**Pull request by DRickard**
* Updated SELECT queries to alias column names (where necessary) to impose camelCase on JSON field names
* Tweaked Institution and Job field-name constants to match values in OpenAPI spec
* Tweaks to db script to synchronize field names (that PostgreSQL actually ignores)

### CODE REVIEW

File: src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java [hunk 1]
```diff
@@ -83,19 +85,25 @@ public class HarvestScheduleStoreServiceImpl implements HarvestScheduleStoreServ
...
      * The insert query for jobs.
      */
     private static final String ADD_JOB =
-            "INSERT INTO public.harvestjobs(institutionID, repositoryBaseURL, metadataPrefix, sets," +
+            "INSERT INTO public.harvestjobs(institutionId, repositoryBaseUrl, metadataPrefix, sets," +
                     " lastSuccessfulRun, scheduleCronExpression) VALUES($1, $2, $3, $4, $5, $6) RETURNING id";
 
     /**
-     * The select query for all jobs.
+     * The select query for all jobs. N.b. awkward format to satisfy checkstyle rules
```
Comment (ksclarke) [inline src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:101]: Is the checkstyle rule here line length or something else?

Comment (DRickard) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:101]: The GET_JOB and LIST__JOBS queries originally had same layout for SELECT sections, and checkstyle was complaining about duplicate string literals.

Comment (ksclarke) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:101]: I'm not sure I follow. One was `"SELECT * FROM public.harvestjobs WHERE id = $1"` and the other `"SELECT * FROM public.harvestjobs ORDER BY institutionID"` which are different string literals(?)

Unrelated then, as a general strategy instead of concatenating string parts into a single string with `+`s, heredocs are supported in Java now:

```
private static final String LIST_JOBS = """
      SELECT institutionId AS \"institutionId\", repositoryBaseUrl AS \"repositoryBaseUrl\",
      metadataPrefix AS \"metadataPrefix\", sets, lastSuccessfulRun AS \"lastSuccessfulRun\",
      scheduleCronExpression AS \"scheduleCronExpression\" FROM public.harvestjobs ORDER BY \"institutionId\"
    """;
```

Comment (ksclarke) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:101]: Oh do you mean in the new form there were string parts that were shared across the queries. I think I see that now. Heredocs would also solve that.

File: src/test/resources/db/prldb.sql [hunk 2]
```diff
@@ -47,8 +47,8 @@ CREATE TABLE public.institutions (
 
 CREATE TABLE public.harvestjobs (
     id SERIAL PRIMARY KEY,
-    institutionID INT,
-    repositoryBaseURL TEXT NOT NULL,
+    institutionId INT,
```
Comment (ksclarke) [inline src/test/resources/db/prldb.sql:50]: Is the upcasing of the last word (something we do in Java) problematic for psql?

Comment (markmatney) [reply src/test/resources/db/prldb.sql:50]: I think I can answer: I had written the OpenAPI schemas without upcasing the last word, and the ticket specified that those schemas should be matched. I'm fine with either form for JSON fields.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:50]: (To directly answer your question @ksclarke: no)

**Review by markmatney — COMMENTED** [review]
No feedback from me at the moment besides agreeing with @ksclarke's comments

**Comment by ksclarke** [comment]
Just a note that the branch is out-of-date now.

File: src/test/resources/db/prldb.sql (hunk 2 above)
Comment (ksclarke) [reply src/test/resources/db/prldb.sql:50]: I personally think consistency between the two would be better if there aren't technical problems with that.

File: src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java (hunk 1 above)
Comment (ksclarke) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:101]: As an aside, I'm kind of curious whether we'd need to use the backslash escaping of quotes with the text block format(?) The JEP SQL example shows just the standalone quotes (no escaping).

File: src/test/resources/db/prldb.sql (hunk 2 above)
Comment (DRickard) [reply src/test/resources/db/prldb.sql:50]: Tweaked db script to make field names consistent

File: src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java (hunk 1 above)
Comment (DRickard) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:109]: Switched to heredocs for long query strings.

File: src/test/resources/db/prldb.sql (hunk 2 above)
Comment (markmatney) [reply src/test/resources/db/prldb.sql:50]: I think what @ksclarke meant is to have all-uppercase acronyms when they appear at the end of a name, like `institutionID`.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:50]: Mine definitely are today 😅

Comment (ksclarke) [reply src/test/resources/db/prldb.sql:50]: 😁

Comment (DRickard) [reply src/test/resources/db/prldb.sql]: Switched to all caps for ID and URL

File: src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java [hunk 3]
```diff
@@ -75,25 +83,36 @@ public class HarvestScheduleStoreServiceImpl implements HarvestScheduleStoreServ
...
-            "INSERT INTO public.harvestjobs(institutionID, repositoryBaseURL, metadataPrefix, sets," +
-                    " lastSuccessfulRun, scheduleCronExpression) VALUES($1, $2, $3, $4, $5, $6) RETURNING id";
+    private static final String ADD_JOB = """
+        INSERT INTO public.harvestjobs(institutionId, repositoryBaseUrl, metadataPrefix, sets,
+        lastSuccessfulRun, scheduleCronExpression) VALUES($1, $2, $3, $4, $5, $6) RETURNING id
+        """;
 
     /**
-     * The select query for all jobs.
+     * The select query for all jobs. N.b. awkward format to satisfy checkstyle rules
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java:109]: Assuming the use of Heredocs changes this:

```suggestion
     * The select query for all jobs.
```

File: src/test/resources/db/prldb.sql [hunk 4]
```diff
@@ -169,7 +169,8 @@ COPY public.harvestjobs (id, institutionID, repositoryBaseURL, metadataPrefix, s
 --
 
 ALTER TABLE ONLY public.harvestjobs
-    ADD CONSTRAINT harvestjobs_fkey FOREIGN KEY(institutionID) REFERENCES public.institutions(id);
+    ADD CONSTRAINT harvestjobs_fkey FOREIGN KEY(institutionID) REFERENCES public.institutions(id)
+    ON DELETE CASCADE;;
```
Comment (markmatney) [inline src/test/resources/db/prldb.sql:173]: Why this change?

Comment (DRickard) [reply src/test/resources/db/prldb.sql:173]: The `ON DELETE CASCADE` causes linked harvestjob records to be deleted when the "parent" institution record is deleted.  This was a specified requirement for the `removeInstitution` method; doing it in the db by the cascade command is faster than via query.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:173]: Okay. I assume that without this change, attempting to delete an institution that has jobs associated with it would cause an error? If so, then I think this should have a separate bugfix PR that also adds a test proving that the schedule store service works as expected in that scenario.

Comment (DRickard) [reply src/test/resources/db/prldb.sql:173]: I agree on the new test. In my experience, without the delete cascade, deleting an institution record would leave orphaned records in the harvetjobs table, which would be an error from our viewpoint but not from the db's viewpoint.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:173]: Even with the foreign key constraint? That seems odd to me.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:173]: Anyway... agreed that it's an issue.

Comment (markmatney) [reply src/test/resources/db/prldb.sql:173]: In addition to it being a separate PR, I would prefer deferring it until after the scheduler service tickets are done, since I am currently in the middle of a major refactor of the schedule store service tests. I think it's fine to let this bug exist until then.

File: src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java (hunk 3 above)
Comment (DRickard) [reply src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java]: Removed obsolete comment

**Review by markmatney — APPROVED** [review]

**Review by ksclarke — APPROVED** [review]
